### PR TITLE
[E2E] Fix Percy flake `parameters-widget`

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Cypress was complaining that the icon "pencil" is still animating when it tries to click on it.
https://docs.cypress.io/guides/references/error-messages#cy-failed-because-the-element-is-currently-animating

Example of a failed run:
https://github.com/metabase/metabase/actions/runs/4251521072/jobs/7394267149#step:10:442

This was happening because we started all mobile tests from the desktop view.
Then we resized the window in the middle of the test.

The actual fix is https://github.com/metabase/metabase/commit/2f019fbbf24e4466de92ad23ff42aecc3af9feba

Explanation:
Using the test config object and setting the viewport size there removes the need for resizing in the middle of the test.

### Additional notes:
- This PR also fixes incorrect test titles

### CODE REVIEW

File: frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+  visitDashboard,
+  editDashboard,
+} from "__support__/e2e/helpers";
 import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
 
 const { PRODUCTS_ID } = SAMPLE_DATABASE;
 
 const questionDetails = {
-  query: { "source-table": PRODUCTS_ID },
+  query: { "source-table": PRODUCTS_ID, limit: 2 },
```
Comment (nemanjaglumac) [inline frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js:11]: This is not directly related to a flake, but it definitely doesn't hurt the overall performance.

When testing locally with the simulated CPU throttling (4x slowdown), tests were timing out because the initial dashboard results loading was too slow!

As a general rule, whenever you can limit the number of results without affecting the quality of the test, do it.

File: frontend/test/__support__/e2e/helpers/e2e-dashboard-helpers.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ export function showDashboardCardActions(index = 0) {
 
 export function editDashboard() {
   cy.icon("pencil").click();
+  cy.findByText("You're editing this dashboard.");
```
Comment (nemanjaglumac) [inline frontend/test/__support__/e2e/helpers/e2e-dashboard-helpers.js:19]: This can only improve the overall stability of tests using this particular helper.

File: frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js [hunk 3]
```diff
@@ -80,41 +84,44 @@ describe(`visual tests > dashboard > parameters widget`, () => {
...
-          "position",
-          "fixed",
-        );
-      });
-
-      it("is sticky in edit mode", () => {
-        cy.viewport(MOBILE_WIDTH, 667);
+    describe(
+      "mobile",
+      {
```
Comment (WiNloSt) [inline frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js:89]: This is awesome. I never knew we could do this 🎉 

Ref: https://docs.cypress.io/guides/references/configuration#Test-specific-Configuration

File: frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js (hunk 1 above)
Comment (WiNloSt) [reply frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js:11]: ⚡

File: frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js [hunk 4]
```diff
@@ -63,8 +67,8 @@ describe(`visual tests > dashboard > parameters widget`, () => {
         );
       });
 
-      it("is sticky in edit mode", () => {
-        cy.icon("pencil").click();
+      it("is not sticky in edit mode", () => {
```
Comment (WiNloSt) [inline frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js:70]: Nice catch 👍

**Review by WiNloSt — APPROVED** [review]
Looks good to me.
